server/activate_hooks.go

```diff
@@ -41,6 +41,10 @@ func (p *Plugin) OnActivate() error {
 		p.API.LogError("Server configuration is not compatible")
 	}
 
+	if err := p.OnConfigurationChange(); err != nil {
```

Comment:
nit (0/5): I understand we are calling OnConfigurationChange here to make sure we exit if we cannot create the channel or user. Nevertheless, I wonder if we should separate this in a different function (e.g. CreateUserAndChannel). Semantically, here there are no configuration changes. We are just trying to create the channel and the user to exit if they cannot be created.

Reply:
While I totally see your point, it just doesn't seam worth it to me to refactor the code to get the 100% correct semantic.

Reply:
@larkox I thought the same thing, and landed on @hanzei's conclusion. However, we don't want to encourage this so I'm making a ticket to change it.